src/test/java/codeu/controller/ProfileServletTest.java

```diff
@@ -0,0 +1,214 @@
+              Instant.now());
+
+    user.setAboutMe("fake about me");
+    Mockito.when(mockUserStore.getUser("test_user")).thenReturn(user);
+    Mockito.when(mockUser.getAboutMe()).thenReturn("fake about me");
+
+    UUID fakeConversationId = UUID.randomUUID();
+    List<Message> fakeMessageList = new ArrayList<>();
+    fakeMessageList.add(
+        new Message(
+            UUID.randomUUID(),
+            fakeConversationId,
+            fakeUserId,
+            "test message",
+            Instant.now()));
+
+    Mockito.when(mockMessageStore.getMessagesByUser(fakeUserId))
+        .thenReturn(fakeMessageList);
+
+    profileServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("about","fake about me");
+    Mockito.verify(mockRequest).setAttribute("currentProfile", "test_user");
+    Mockito.verify(mockRequest).setAttribute("messages",fakeMessageList);
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoGet_noAboutMe() throws IOException, ServletException {
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/user/test_user");
+
+    UUID fakeUserId = UUID.randomUUID();
+    User user =
+          new User(
+              fakeUserId,
+              "test_user",
+              "$2a$10$.e.4EEfngEXmxAO085XnYOmDntkqod0C384jOR9oagwxMnPNHaGLa",
+              Instant.now());
+
+    Mockito.when(mockUserStore.getUser("test_user")).thenReturn(user);
+
+    UUID fakeConversationId = UUID.randomUUID();
+    List<Message> fakeMessageList = new ArrayList<>();
+    fakeMessageList.add(
+        new Message(
+            UUID.randomUUID(),
+            fakeConversationId,
+            fakeUserId,
+            "test message",
+            Instant.now()));
+
+    Mockito.when(mockMessageStore.getMessagesByUser(fakeUserId))
+        .thenReturn(fakeMessageList);
+
+    profileServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("about","This user hasn't made a profile yet :(");
+    Mockito.verify(mockRequest).setAttribute("currentProfile", "test_user");
+    Mockito.verify(mockRequest).setAttribute("messages",fakeMessageList);
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoGet_nullUser() throws IOException, ServletException {
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/user/test_user");
+
+    Mockito.when(mockUserStore.getUser("test_user")).thenReturn(null);
+
+    profileServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("error","THAT USER DOESN'T EXIST! ENTER A VALID USERNAME");
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoGet_noSentMessages()  throws IOException, ServletException {
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/user/test_user");
+
+    UUID fakeUserId = UUID.randomUUID();
```

Comment:
You created fakeUser multiple times, is there a reason you didn't just do it in the setup once?

Reply:
I did that because some of the tests don't require a fakeUser object to be made. For example, testDoGet_nullUser() relies on there being no user made, this the user attribute being a null value.